### PR TITLE
bugfix/15170-stocktools-popup-lang

### LINKED ISSUE #15170: StockTools: Missing lang options for `<indicator>.params`
#### Expected behaviour
Available options to change default `params` names, like `period` or `index`.

#### Actual behaviour
We display the name of the option:
Screenshot 2021-02-18 at 12 11 51

List of options:
- `index`
- `period`
- `standardDeviation`
- `periodTenkan`
- `periodSenkouSpanB`
- `periodATR`
- `multiplierATR`
- `shortPeriod`
- `longPeriod`
- `signalPeriod`
- `decimals`
- `algorithm`
- `topBand`
- `bottomBand`
- `initialAccelerationFactor`
- `maxAccelerationFactor`
- `increment`
- `multiplier`
- `ranges`
- `highIndex`
- `lowIndex`
- `deviation`
- `xAxisUnit`
- `factor`

#### Live demo with steps to reproduce
https://www.highcharts.com/demo/stock/stock-tools-gui

Note: Options work when defined in `lang`: https://jsfiddle.net/BlackLabel/jpyeu2mh/ (choose IKH indicator) - but default options are not defined (thus not listed in API: https://api.highcharts.com/highstock/lang.navigation.popup )

Steps:
- click on the Indicators button
- check some of the indicators (MACD, ZigZag etc)

### PR DESCRIPTION
**Pull request by raf18seb**
Fixed #15170, added `lang` options for indicators parameters in Stock Tools popup.

### CODE REVIEW

**Review by pawelfus — APPROVED** [review]
Thanks 👍

**Review by pawelfus — CHANGES_REQUESTED** [review]
Sorry, prematurely accepted. Could move lang definitions from `Popup` to `StockToolsGUI`, where the rest of lang options are set? See: https://github.com/highcharts/highcharts/blob/master/ts/Stock/StockToolsGui.ts#L249

File: ts/Extensions/Annotations/Popup.ts [hunk 1]
```diff
@@ -165,6 +166,41 @@ var indexFilter = /\d/g,
     LI = 'li',
     H3 = 'h3';
 
+// Lang definitions for indicators parameters in Stock Tools popup, #15170
+setOptions({
```
Comment (karolkolodziej) [inline ts/Extensions/Annotations/Popup.ts:170]: Rather question than request.
Is there any difference in performance or anything else between `setOptions` and `merge` like below?
```
merge(defaultOptions, {
    lang: {
       ...
    }
})
```

**Review by karolkolodziej — APPROVED** [review]

File: ts/Extensions/Annotations/Popup.ts (hunk 1 above)
Comment (raf18seb) [reply ts/Extensions/Annotations/Popup.ts]: Yes. `setOptions` changes (extends) the first argument when merge only returns the whole new object (without assigning or changing the first one). So you would have to either do something like this:
```js
defaultOptions = merge(defaultOptions, {
    lang: {
       ...
    }
})
```
or
```js
merge(true, defaultOptions, {
    lang: {
       ...
    }
})
```

https://api.highcharts.com/class-reference/Highcharts#.merge%3CT%3E

Comment (karolkolodziej) [reply ts/Extensions/Annotations/Popup.ts]: Ok, `setOptions` looks correctly in this case! Thanks ;)

**Review by TorsteinHonsi — APPROVED** [review]
Thanks! Pending @pawelfus' final approval

**Review by pawelfus — APPROVED** [review]
Looks great, thanks!

**Pull request merged**